source/base/hdf5.cc

```diff
@@ -588,13 +601,17 @@ namespace HDF5
     , global_no_collective_cause(H5D_MPIO_SET_INDEPENDENT)
   {
     hdf5_reference = std::shared_ptr<hid_t>(new hid_t, [](hid_t *pointer) {
-      // Relase the HDF5 resource
-      H5Dclose(*pointer);
+      // Release the HDF5 resource
+      const herr_t ret = H5Dclose(*pointer);
+      Assert(ret >= 0, ExcInternalError());
+      (void)ret;
```

Comment:
I believe these are custom deleters, so we should use `AssertNothrow` here and below.

Reply:
Good point! I fixed all the places I could find in that file.